code/modules/admin/admin_verbs.dm

```diff
@@ -526,3 +527,49 @@ var/list/admin_verbs_hideable = list(
 	if(holder)
 		src.holder.output_ai_laws()
 
+/client/proc/readmin()
+	set name = "Re-admin self"
+	set category = "Admin"
+	set desc = "Regain your admin powers."
+	var/list/rank_names = list()
+	for(var/datum/admin_rank/R in admin_ranks)
+		rank_names[R.name] = R
+	var/datum/admins/D = admin_datums[ckey]
+	var/rank = null
+	if(config.admin_legacy_system)
+		//load text from file
+		var/list/Lines = file2list("config/admins.txt")
+		for(var/line in Lines)
+			var/list/splitline = text2list(line, " = ")
+			if(splitline[1] == ckey)
+				if(splitline.len >= 2)
+					rank = ckeyEx(splitline[2])
+				break
+			continue
+	else
+		if(!dbcon.IsConnected())
+			message_admins("Warning, mysql database is not connected.")
+			src << "Warning, mysql database is not connected."
+			return
+		var/DBQuery/query = dbcon.NewQuery("SELECT ckey, rank FROM [format_table_name("admin")] WHERE ckey = '[ckey]'")

```

Comment:
Still best to sanitize the ckey first, imo.

On Fri, Jan 2, 2015 at 6:49 PM, tkdrg <notifications@github.com> wrote:

> In code/modules/admin/admin_verbs.dm
> <https://github.com/tgstation/-tg-station/pull/6743#discussion-diff-22416558>
> :
>
> > +	if(config.admin_legacy_system)
> > +		//load text from file
> > +		var/list/Lines = file2list("config/admins.txt")
> > +		for(var/line in Lines)
> > +			var/list/splitline = text2list(line, " = ")
> > +			if(splitline[1] == ckey)
> > +				if(splitline.len >= 2)
> > +					rank = ckeyEx(splitline[2])
> > +				break
> > +			continue
> > +	else
> > +		if(!dbcon.IsConnected())
> > +			message_admins("Warning, mysql database is not connected.")
> > +			src << "Warning, mysql database is not connected."
> > +			return
> > +		var/DBQuery/query = dbcon.NewQuery("SELECT ckey, rank FROM [format_table_name("admin")] WHERE ckey = '[ckey]'")
>
> You told me ckeys are alphanumeric, nexis. Please don't tell me you lied.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/6743/files#r22416558>.
>